poetry/puzzle/provider.py

```diff
@@ -815,6 +815,7 @@ def progress(self):
 
         self._in_progress = False
 
+    @classmethod
     def _execute_setup(self):
```

Comment:
It doesn't seem you are using the class itself. In this case it is better to use `staticmethod`:
```suggestion
    @staticmethod
    def _execute_setup():
```

Reply:
this one wasn't me, I am not sure what to do here, 

https://github.com/python-poetry/poetry/blob/9f93df1140efd44f37d07fc04fe5eb9abdc105db/poetry/puzzle/provider.py#L329

Master is broken right now.